app/forms/state_file/esign_declaration_form.rb

```diff
@@ -3,15 +3,26 @@ class EsignDeclarationForm < QuestionsForm
     set_attributes_for :state_file_efile_device_info, :device_id
     set_attributes_for :intake,
                        :primary_esigned,
-                       :spouse_esigned
+                       :spouse_esigned,
+                       :primary_signature_pin,
+                       :spouse_signature_pin
 
     validates :primary_esigned, acceptance: { accept: 'yes', message: ->(_object, _data) { I18n.t("views.ctc.questions.confirm_legal.error") }}
     validates :spouse_esigned, acceptance: { accept: 'yes', message: ->(_object, _data) { I18n.t("views.ctc.questions.confirm_legal.error") }}, if: -> { @intake.ask_spouse_esign? }
     validate :validate_intake_already_submitted
+    validates :primary_signature_pin, presence: true, signature_pin: true, if: -> { @intake.ask_for_signature_pin? }
+    validates :spouse_signature_pin, presence: true, signature_pin: true, if: -> { @intake.ask_for_signature_pin?  && @intake.spouse.full_name.present? }
 
     def save
       return false unless valid?
       attrs = @intake.ask_spouse_esign? ? attributes_for(:intake) : attributes_for(:intake).except(:spouse_esigned)
+      attrs = @intake.ask_for_signature_pin? ? attrs : attrs.except(:primary_signature_pin, :spouse_signature_pin)
```

Comment:
I might be misreading, but I don't think we need this line? It seems to be covered by the lines directly under it.